### DISCUSSION TITLE
next router vs next/dist/client/router?

### DISCUSSION THREAD

**Discussion by Vandivier** [vercel/next.js#44160] [Help]
what's the difference? is there a best practice or cases where i should prefer one to the other? thanks!

![image](https://user-images.githubusercontent.com/5559355/208489895-ea47b015-f1c8-453b-a16f-7cee3c4c3311.png)

**Comment by icyJoseph** [comment]
**_Use the `next/router` import._**

There's cases where you might want to reach out for a type, or util buried within a package. Then one might do `<package>/dist` or any other form. 

For example:

```ts
import type {NextApiRequestCookies, NextApiRequestQuery} from 'next/dist/server/api-utils';
```

Assuming that for some wicked reason I'd ever need access to those. 

---

In this case I think it is odd that the quick fix is suggesting that.

If we see the contents of `node_modules/next/router.js`:

```js
module.exports = require('./dist/client/router')
```

We see it is a re-export of the file produced when next.js publishes it's versions, roughly speaking.

If you are not so familiar with package publishing, bundling tools and CJS vs ESM, do not overthink this part and simply import `next/router`. 

For any other situation, do as  the docs say, https://nextjs.org/docs/api-reference/next/router#userouter for example this page clearly shows `next/router`.

**Comment by TTTerrence** [comment]
您好，您发来的邮件杨珊已经收到，我会尽快处理和回复，谢谢！       Thank you. Your email is received and will be handled  as soon as possible.
